ord_schema/units.py

```diff
@@ -294,7 +294,7 @@ def resolve(self, string: str, allow_range: bool = False) -> ord_schema.UnitMess
             return message(value=value, precision=precision, units=unit)
         return message(value=value, units=unit)
 
-    def resolve_unit(self, string_unit: str) -> Tuple[ord_schema.UnitMessage, int]:
+    def resolve_unit(self, string_unit: str) -> Tuple[ord_schema.UnitMessage, ord_schema.Message]:
```

Comment:
I thought this was correct: the type check fail actually comes from `unit_synonyms` in `__init__` (the second key should be an enum value).

Reply:
Could you clarify? All tests are passing but I will wait for you to resolve this comment (or help me figure out what's wrong) before merging

Reply:
From this [line](https://github.com/open-reaction-database/ord-schema/blob/d23c1c3a7acf40b11ca8c8bca61ab8f625c4c67e/ord_schema/units.py#L220C9-L220C113) the second key in `unit_synonyms` is hinted to be an `ord_schema.Message` (causing the "wrong" return type in `resolve_unit`), but it should be just an ENUM option. So maybe it's better to change type hints there. It doesn't matter much tho.